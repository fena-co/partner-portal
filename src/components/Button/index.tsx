import styled, { css } from 'styled-components';
import { FC } from 'react';
import Spinner from '../Spinner';

const ButtonContained = css`
  background-color: #13273f;
  color: #ffffff;
  border: none;
  font-weight: 600;
  &:hover {
    background-color: #19385e;
    box-sizing: border-box;
    filter: drop-shadow(0px 7px 20px rgba(108, 108, 138, 0.6));
  }
  &:disabled {
    box-sizing: border-box;
    background: #dbe3eb;
    filter: none;
  }
`;

const ButtonOutlined = css`
  background-color: #ffffff;
  font-weight: 600;
  border: 1px solid #dbe3eb;
  color: #0e233e;
  &:hover {
    box-sizing: border-box;
    border: 1px solid #13273f;
  }
  &:disabled {
    border: 1px solid #dbe3eb;
    box-sizing: border-box;
    color: #dbe3eb;
  }
`;

const FullWidth = css`
  width: 100%;
`;

const Button = styled.button<{ fullWidth?: boolean; variant: string }>`
  font-family: Montserrat;
  font-style: normal;
  font-size: 15px;
  line-height: 18px;
  text-align: center;
  border-radius: 5px;
  // padding-top: 18px;
  // padding-bottom: 18px;
  padding: 12px 21px;
  cursor: pointer;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  ${({ variant }) =>
    variant === `contained` ? ButtonContained : ButtonOutlined}
  ${({ fullWidth }) => (fullWidth ? FullWidth : '')}
`;

const ButtonWithChildren: FC<any> = ({
  children,
  loading,
  variant,
  ...rest
}) => {
  if (loading) {
    return (
      <Button disabled variant={variant} {...rest}>
        <Spinner />
        Processing
      </Button>
    );
  }

  return (
    <Button variant={variant} {...rest}>
      {children}
    </Button>
  );
};

export default ButtonWithChildren;
