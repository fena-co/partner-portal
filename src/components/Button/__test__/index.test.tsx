import Button from '../index';
import renderer from 'react-test-renderer';

describe('Button Block', () => {
  it('renders correctly with variant contained ', () => {
    const tree = renderer
      .create(<Button variant={`contained`}>Submit</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with variant outlined ', () => {
    const tree = renderer
      .create(<Button variant={`outlined`}>Submit</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
