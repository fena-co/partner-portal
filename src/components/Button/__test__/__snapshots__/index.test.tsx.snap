// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Block renders correctly with variant contained  1`] = `
.c0 {
  font-family: Montserrat;
  font-style: normal;
  font-size: 15px;
  line-height: 18px;
  text-align: center;
  border-radius: 5px;
  padding: 12px 21px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  background-color: #13273f;
  color: #ffffff;
  border: none;
  font-weight: 600;
}

.c0:hover {
  background-color: #19385e;
  box-sizing: border-box;
  -webkit-filter: drop-shadow(0px 7px 20px rgba(108,108,138,0.6));
  filter: drop-shadow(0px 7px 20px rgba(108,108,138,0.6));
}

.c0:disabled {
  box-sizing: border-box;
  background: #dbe3eb;
  -webkit-filter: none;
  filter: none;
}

<button
  className="c0"
>
  Submit
</button>
`;

exports[`Button Block renders correctly with variant outlined  1`] = `
.c0 {
  font-family: Montserrat;
  font-style: normal;
  font-size: 15px;
  line-height: 18px;
  text-align: center;
  border-radius: 5px;
  padding: 12px 21px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  background-color: #ffffff;
  font-weight: 600;
  border: 1px solid #dbe3eb;
  color: #0e233e;
}

.c0:hover {
  box-sizing: border-box;
  border: 1px solid #13273f;
}

.c0:disabled {
  border: 1px solid #dbe3eb;
  box-sizing: border-box;
  color: #dbe3eb;
}

<button
  className="c0"
>
  Submit
</button>
`;
